### PR TITLE
fix: Remove hardcoded logo mimetype

### PR DESCRIPTION
**Pull request by ywei390**
## Description

Work Item ID: [AB#34396](https://dev.azure.com/CluedIn-io/c054b4ae-1dab-43c2-af97-3683c744782f/_workitems/edit/34396)

## How has it been tested? 
Manually in local

### CODE REVIEW

File: src/WebExternalSearchProvider.cs [hunk 1]
```diff
@@ -177,27 +179,42 @@ public IEnumerable<Clue> BuildClues(ExecutionContext context, IExternalSearchQue
...
+            var tempUri = string.Empty;
+
+            if (orgWebSite.Logo != null && orgWebSite.Logo.ToString().StartsWith("/"))
+            {
+                tempUri = orgWebSite.Logo.ToString()[1..];
+            }
+
+            if (!string.IsNullOrEmpty(tempUri) && !Uri.IsWellFormedUriString(tempUri, UriKind.Absolute))
+            {
+                orgWebSite.Logo = new Uri(orgWebSite.ResponseUri.AbsoluteUri + tempUri);
```
Comment (andrew-hardy) [inline src/WebExternalSearchProvider.cs:191]: You should be able to use new Uri(Uri, string) as this will handle relative and absolute for you

**Review by andrew-hardy — CHANGES_REQUESTED** [review]

File: src/WebExternalSearchProvider.cs (hunk 1 above)
Comment (ywei390) [reply src/WebExternalSearchProvider.cs]: updated the code, removed unnecessary checking on uri

**Review by andrew-hardy — APPROVED** [review]

File: src/WebExternalSearchProvider.cs [hunk 2]
```diff
@@ -179,25 +179,33 @@ public IEnumerable<Clue> BuildClues(ExecutionContext context, IExternalSearchQue
...
-                using (var stream = client.OpenRead(orgWebSite.Logo)) //Get Full Quality Image
+                if (!string.IsNullOrEmpty(orgWebSite.Logo?.ToString()) && !Uri.IsWellFormedUriString(orgWebSite.Logo?.ToString(), UriKind.Absolute))
                 {
-                    var inArray = StreamUtilies.ReadFully(stream);
-                    if (inArray != null)
+                    orgWebSite.Logo = new Uri(new Uri(orgWebSite.ResponseUri.AbsoluteUri), orgWebSite.Logo.ToString());
+                }
+
+                var client = new RestClient();
+                var req = new RestRequest(orgWebSite.Logo?.ToString(), Method.GET);
```
Comment (andrew-hardy) [inline src/WebExternalSearchProvider.cs:188]: Why would we want to do a HTTP request if orgWebSite.Logo is null?

File: src/WebExternalSearchProvider.cs [hunk 3]
```diff
@@ -179,25 +179,33 @@ public IEnumerable<Clue> BuildClues(ExecutionContext context, IExternalSearchQue
 
             try
             {
-                using (var client = new WebClient())
-                using (var stream = client.OpenRead(orgWebSite.Logo)) //Get Full Quality Image
+                if (!string.IsNullOrEmpty(orgWebSite.Logo?.ToString()) && !Uri.IsWellFormedUriString(orgWebSite.Logo?.ToString(), UriKind.Absolute))
                 {
-                    var inArray = StreamUtilies.ReadFully(stream);
-                    if (inArray != null)
+                    orgWebSite.Logo = new Uri(new Uri(orgWebSite.ResponseUri.AbsoluteUri), orgWebSite.Logo.ToString());
```
Comment (andrew-hardy) [inline src/WebExternalSearchProvider.cs:184]: I don't think this needs to be conditional on orgWebSite.Logo being relative as new Uri will handle both absolute and relative paths

**Review by andrew-hardy — CHANGES_REQUESTED** [review]

**Review by andrew-hardy — DISMISSED** [review]

**Review by andrew-hardy — APPROVED** [review]

**Review by andrew-hardy — APPROVED** [review]

**Pull request merged**